crates/forge/bin/main.rs

```diff
@@ -101,47 +163,96 @@ fn run() -> Result<()> {
             Ok(())
         }
         ForgeSubcommand::Snapshot(cmd) => {
+            command_name = "snapshot";
             if cmd.is_watch() {
                 utils::block_on(watch::watch_gas_snapshot(cmd))
             } else {
                 utils::block_on(cmd.run())
             }
         }
         ForgeSubcommand::Fmt(cmd) => {
+            command_name = "fmt";
             if cmd.is_watch() {
                 utils::block_on(watch::watch_fmt(cmd))
             } else {
                 cmd.run()
             }
         }
-        ForgeSubcommand::Config(cmd) => cmd.run(),
-        ForgeSubcommand::Flatten(cmd) => cmd.run(),
-        ForgeSubcommand::Inspect(cmd) => cmd.run(),
-        ForgeSubcommand::Tree(cmd) => cmd.run(),
+        ForgeSubcommand::Config(cmd) => {
+            command_name = "config";
+            cmd.run()
+        }
+        ForgeSubcommand::Flatten(cmd) => {
+            command_name = "flatten";
+            cmd.run()
+        }
+        ForgeSubcommand::Inspect(cmd) => {
+            command_name = "inspect";
+            cmd.run()
+        }
+        ForgeSubcommand::Tree(cmd) => {
+            command_name = "tree";
+            cmd.run()
+        }
```

Comment:
Please move the entire match block for obtaining the command name to a separate function or block. We strive to keep upstream code as-in as much as possible to avoid merge conflicts when syncing with the upstream branch.

Reply:
Yeah, I thought about that, but then there is a risk when new command is added we forget to add it to that function. If that will simplify merging upstream I can do it though.

Reply:
That wouldn't happen, rust's exhaustiveness check would catch it under `match` if a new command is added.